Adding stats for aborted job count in batch router

sivashanmukh opened the pull request:
**Fixes** # (*issue*)

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

SumanthPuram reviewed (commented):
Instead of doing it here, might be better to do it in recordUploadStatus function? That's where we are tracking event deliveries too.

sivashanmukh commented:
@SumanthPuram There is another setJobStatus [here](https://github.com/rudderlabs/rudder-server/blob/fae990229522e6eda04b7b780ded81ddff998bde/router/batchrouter/batchrouter.go#L606). We won't count those if we move it inside recordUploadStat. @psrikanth88 should we panic on [this error](https://github.com/rudderlabs/rudder-server/blob/fae990229522e6eda04b7b780ded81ddff998bde/router/batchrouter/batchrouter.go#L605)?

SumanthPuram commented:
I think the handling for jobsBySource needs to be improved.

[Here](https://github.com/rudderlabs/rudder-server/blob/fae990229522e6eda04b7b780ded81ddff998bde/router/batchrouter/batchrouter.go#L573) we mark `executing` irrespective of whether src/destination config is present. 

[Here](https://github.com/rudderlabs/rudder-server/blob/fae990229522e6eda04b7b780ded81ddff998bde/router/batchrouter/batchrouter.go#L606) we mark `failed` even if the config is not present.

We should search for the source in the config in both places. If we don't find this in the first, we should not even mark it `executing`

sivashanmukh commented:
I have cleaned it up a little more. This time I moved out the aborted check, consolidated success and fail counts into the same single metric with different tags to differentiate. Regarding updating the job statuses in a better way. Let's take it as a different task.

SumanthPuram commented on router/batchrouter/batchrouter.go:
```diff
@@ -147,14 +147,22 @@ type ErrorResponseT struct {
+		"isWarehouse": fmt.Sprintf("%t", isWarehouse),
+	}
+	for _, paramFilter := range parameterFilters {
+		tags[fmt.Sprintf("param_%s", paramFilter.Name)] = paramFilter.Value
```
The parameter filter also has received_at as a key and the value of received_at timestamp. Timestamps will blow up influx.

Eg
```
{ 
  "source_id": "s1",
  "destination_id": "d1",
  "received_at": "t1"
}
```

Instead of sending all parameter filters, let's send only source and destinations? 
We should use this function for representing any `source` or `destination` in tags.

`misc.GetTagName(destination.Destination.ID, destination.Destination.Name)`

Also the tag names should `source` and `destination` to maintain parity across all metrics.

sivashanmukh replied:
In this context, parameter filters doesn't have received_at. It has only one parameter and that is destId

```
	parameterFilters := []jobsdb.ParameterFilterT{
		{
			Name:     "destination_id",
			Value:    batchJobs.BatchDestination.Destination.ID,
			Optional: false,
		},
	}

```
In any case, I have removed the paramFilter piece and am adding source and destination ids like you suggested. Take a look. Should we add source/dest names here for better readability?

SumanthPuram reviewed (commented):
Need to fix the tags.
